### ISSUE TITLE
Active rules enabled - Custom rules ignored?

### ISSUE THREAD

**Issue by shellshocker** [ukanth/afwall#579]
Since AFWall+ 2.5.2 I have problems when active rules are enabled. After switchting WLAN On/Off my Custom Script rules are ignored.

Here is my Custom Script:

```
##
####################
# Tweaks           #
####################
##

## Kernel
# Disable IPv6
echo 1 > /proc/sys/net/ipv6/conf/all/disable_ipv6
echo 1 > /proc/sys/net/ipv6/conf/default/disable_ipv6
# Privacy IPv6 Address
echo 2 > /proc/sys/net/ipv6/conf/all/use_tempaddr
echo 2 > /proc/sys/net/ipv6/conf/default/use_tempaddr
## System
# Disable Captive Portal (Connect to http://clients3.google.com via UID 1000) 
settings put global captive_portal_detection_enabled 0 
settings put global captive_portal_server localhost
# Disable Global NTP Server 
settings put global ntp_server 127.0.0.1

IPTABLES=/system/bin/iptables
IP6TABLES=/system/bin/ip6tables 

####################
# Purge/Flush      #
####################

# All 'afwall' rules/chains gets flushed automatically, before the custom script is executed
# Flush/Purge all rules expect OUTPUT
$IPTABLES -F INPUT
$IPTABLES -F FORWARD
$IPTABLES -t nat -F
$IPTABLES -t mangle -F
$IP6TABLES -F INPUT
$IP6TABLES -F FORWARD
$IP6TABLES -t nat -F
$IP6TABLES -t mangle -F

# Flush/Purge all chains  
$IPTABLES -X   
$IPTABLES -t nat -X  
$IPTABLES -t mangle -X  
$IP6TABLES -X   
$IP6TABLES -t nat -X  
$IP6TABLES -t mangle -X

####################
# Defaults         #
####################

# IPv4 connections
$IPTABLES -P INPUT   DROP
$IPTABLES -P FORWARD DROP
$IPTABLES -P OUTPUT  DROP

# IPv6 connections  
$IP6TABLES -P INPUT DROP  
$IP6TABLES -P FORWARD DROP  
$IP6TABLES -P OUTPUT DROP

#####################
# Special Rules     #
#####################

# Allow loopback interface lo 
$IPTABLES -A INPUT -i lo -j ACCEPT 
$IPTABLES -A "afwall" -o lo -j ACCEPT

# Force a specific NTP (ntp0.fau.de), Location: University Erlangen-Nuernberg
$IPTABLES -t nat -A OUTPUT -p tcp --dport 123 -j DNAT --to-destination 131.188.3.222:123
$IPTABLES -t nat -A OUTPUT -p udp --dport 123 -j DNAT --to-destination 131.188.3.222:123

#####################
# Incoming Traffic  #
#####################

# Allow ICMP packets
$IPTABLES -A INPUT -p icmp -m icmp --icmp-type echo-reply -j ACCEPT 
$IPTABLES -A INPUT -p icmp -m icmp --icmp-type echo-request -j ACCEPT 
$IPTABLES -A INPUT -p icmp -m icmp --icmp-type destination-unreachable -j ACCEPT

# Allow all traffic from an established connection
$IPTABLES -A INPUT -m state --state ESTABLISHED,RELATED -j ACCEPT

####################
# Outgoing Traffic #
####################

## Block Facebook ASN
$IPTABLES -A "afwall" -d 31.13.24.0/21 -j REJECT
$IPTABLES -A "afwall" -d 31.13.64.0/18 -j REJECT
$IPTABLES -A "afwall" -d 45.64.40.0/22 -j REJECT
$IPTABLES -A "afwall" -d 66.220.144.0/20 -j REJECT
$IPTABLES -A "afwall" -d 69.63.176.0/20 -j REJECT
$IPTABLES -A "afwall" -d 69.171.224.0/19 -j REJECT
$IPTABLES -A "afwall" -d 74.119.76.0/22 -j REJECT
$IPTABLES -A "afwall" -d 103.4.96.0/22 -j REJECT
$IPTABLES -A "afwall" -d 157.240.0.0/17 -j REJECT
$IPTABLES -A "afwall" -d 173.252.64.0/18 -j REJECT
$IPTABLES -A "afwall" -d 179.60.192.0/22 -j REJECT
$IPTABLES -A "afwall" -d 185.60.216.0/22 -j REJECT
$IPTABLES -A "afwall" -d 204.15.20.0/22 -j REJECT
```

So what happens? Well, when active rules are enabled I can surf to www.facebook.com after switching WLAN off and on again. After manually apply firewall rules Facebook gets blocked again. When I disable active rules, everything is working - even after switchting WLAN off and on again.

Ideas? Bug?

**Comment by Superwallah** [comment]
Hi Shellshocker. Is this issue still relevant or can we close it? Can you provide a log that shows the block, please?

**Comment by clocken** [comment]
Hi @ukanth, sorry for raising the dead but I can confirm that this issue still exists. Custom rules are indeed ignored when *Active rules* is enabled and the (wifi) connection changes. I have created a PR that fixes this - see #1253 for a detailed analysis of why this is the case.

I'm on Fairphone 3 with LineageOS 18.1
AFWall+ 3.5.2.1 (from F-Droid)
